Log git hash during startup

rschu1ze opened the pull request:
The ClickHouse server currently logs at startup a compiler-generated "build id" which is different for each build. That makes it useless to determine the exact source code state in tests (e.g. BC test) or in user log files (e.g. if someone compiled an intermediate version of ClickHouse).

This PR adds the Git hash of the build to the log output.

Current log message:
  `Starting ClickHouse 22.10.1.1 with revision 54467, build id: 6F35820328F89C9F36E91C447FF9E61CAF0EF019, PID 42633`

New log message:
  `Starting ClickHouse 22.10.1.1 (revision 54467, git hash: b6b1f7f763f94ffa12133679a6f80342dd1c3afe, build id: 47B12BE61151926FBBD230DE42F3B7A6652AC482), PID 981813`

The Git hash is already available in view SYSTEM.BUILD_OPTIONS. This PR reuses some bits of the logic which embeds the Git hash into the build (it makes it available to more places in the code).

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

rschu1ze committed:
Log git hash during startup

We currently only log a compiler-generated "build id" at startup which
is different for each build. That makes it useless to determine the
exact source code state in tests (e.g. BC test) and from user log files
(e.g. if someone compiled an intermediate version of ClickHouse).

Current log message:
  Starting ClickHouse 22.10.1.1 with revision 54467, build id: 6F35820328F89C9F36E91C447FF9E61CAF0EF019, PID 42633

New log message:
  Starting ClickHouse 22.10.1.1 (revision 54467, git hash: b6b1f7f763f94ffa12133679a6f80342dd1c3afe, build id: 47B12BE61151926FBBD230DE42F3B7A6652AC482), PID 981813

rschu1ze committed:
Fix build

rschu1ze committed:
Fix standalone keeper build

alexey-milovidov commented on CMakeLists.txt:
```diff
@@ -18,7 +18,44 @@ include (cmake/target.cmake)
 include (cmake/clang_tidy.cmake)
-include (cmake/git_status.cmake)
+
+find_package(Git)
```
We already have it in the system.build_options, let's reuse that info.

rschu1ze committed:
Simplify a bit

evillique commented on CMakeLists.txt:
```diff
@@ -18,7 +18,44 @@ include (cmake/target.cmake)
 include (cmake/tools.cmake)
 include (cmake/ccache.cmake)
 include (cmake/clang_tidy.cmake)
-include (cmake/git_status.cmake)
```
Why don't we just add these changes to the `.cmake` file and not bring it into the main `CMakeLists.txt`?

rschu1ze replied:
Done

rschu1ze committed:
Move git calls back into git.cmake

+ renamed the file from originally "git_status.cmake" to "git.cmake"
  (because we not longer run only "git status")

rschu1ze committed:
Cosmetics
